WordPress/Classes/ViewRelated/SuggestionsListViewModel.swift

```diff
@@ -123,15 +123,42 @@ import CoreData
         }
         switch suggestions {
         case .users(let userSuggestions):
+            let sortedUserSuggestions = Self.sort(
+                userSuggestions: userSuggestions, by: searchQuery
+            )
             return Self.searchResultByMovingProminentSuggestionsToTop(
-                userSuggestions: userSuggestions,
+                userSuggestions: sortedUserSuggestions,
                 prominentSuggestionsIds: prominentSuggestionsIds
             )
         case .sites(let siteSuggestions):
             return .sites(siteSuggestions)
         }
     }
 
+    /// Sort user suggestions by prefix first, then alphabetically. The collection is sorted first by checking if displayName begins with the provided prefix. The remaining items are sorted alphabetically by displayName. The prefix comparison is both case-insensitive and diacritic-insensitive.
+    /// - Parameter userSuggestions: The user suggestions collection to be sorted.
+    /// - Parameter prefix: The prefix to be used when checking the usernames and displayNames.
+    /// - Returns: The sorted user suggestion collection.
+    private static func sort(userSuggestions: [UserSuggestion], by prefix: String) -> [UserSuggestion] {
+        guard !userSuggestions.isEmpty, !prefix.isEmpty else { return userSuggestions }
+        var prefixedUserSuggestions = [UserSuggestion]()
+        var otherUserSuggestions = [UserSuggestion]()
+
+        for suggestion in userSuggestions {
+            guard let username = suggestion.username else {
+                continue
+            }
+            if let displayName = suggestion.displayName,
+               (username.beginsWith(prefix: prefix) || displayName.beginsWith(prefix: prefix)) {
+                prefixedUserSuggestions.append(suggestion)
+            } else {
+                otherUserSuggestions.append(suggestion)
+            }
+        }
+
+        return prefixedUserSuggestions.sorted() + otherUserSuggestions.sorted()
+    }
```

Comment:
nitpick: May I suggest a simpler and more compact implementation, using `Dictionary(grouping:by:)`?

```suggestion
    private static func sort(userSuggestions: [UserSuggestion], by prefix: String) -> [UserSuggestion] {
        guard !userSuggestions.isEmpty, !prefix.isEmpty else { return userSuggestions }
        let triagedList = Dictionary(grouping: userSuggestions) { suggestion in
            suggestion.username?.beginsWith(prefix: prefix) == true || suggestion.displayName?.beginsWith(prefix: prefix) == true
        }
        return (triagedList[true] ?? []).sorted() + (triagedList[false] ?? []).sorted()
    }
```

Reply:
That's awesome, I didn't know about `Dictionary(grouping:by:)`. Thanks for sharing! I will apply your suggestion.